sonar-db/src/main/java/org/sonar/db/purge/IssueRemovedListener.java

```diff
@@ -21,21 +21,7 @@
 
 import java.util.List;
 
-public interface PurgeListener {
-
-  PurgeListener EMPTY = new PurgeListener() {
-    @Override
-    public void onComponentDisabling(String uuid) {
-      // do nothing
-    }
-
-    @Override
-    public void onIssuesRemoval(String projectUuid, List<String> issueKeys) {
-      // do nothing
-    }
-  };
-
-  void onComponentDisabling(String uuid);
+public interface IssueRemovedListener {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "IssueRemovedListener" interface with the @FunctionalInterface annotation [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)